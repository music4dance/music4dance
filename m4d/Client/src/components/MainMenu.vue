<template>
  <div>
    <b-alert v-if="isTest" show variant="warning" style="margin-bottom: 0"
      >This is a TEST site. Please navigate to
      <a href="https://www.music4dance.net" class="alert-link"
        >wwww.music4dance.net</a
      >
      to use the production site.</b-alert
    >
    <b-alert
      v-if="context.updateMessage"
      show
      variant="danger"
      style="margin-bottom: 0"
      >{{ context.updateMessage }}</b-alert
    >
    <b-navbar id="mainMenu" type="dark" variant="primary" toggleable="lg" fixed>
      <b-navbar-brand href="/">
        <img src="/images/header-logo.png" height="40" title="music4dance" />
      </b-navbar-brand>

      <b-navbar-toggle target="nav-collapse"></b-navbar-toggle>

      <b-collapse id="nav-collapse" is-nav>
        <b-navbar-nav>
          <b-nav-item-dropdown id="music-menu" text="Music">
            <b-dropdown-item href="/dances">Dances</b-dropdown-item>
            <b-dropdown-item
              href="/dances/ballroom-competition-categories"
              class="nav-subitem"
              >Ballroom</b-dropdown-item
            >
            <b-dropdown-item href="/dances/latin" class="nav-subitem"
              >Latin</b-dropdown-item
            >
            <b-dropdown-item href="/dances/swing" class="nav-subitem"
              >Swing</b-dropdown-item
            >
            <b-dropdown-item href="/dances/tango" class="nav-subitem"
              >Tango</b-dropdown-item
            >
            <b-dropdown-item href="/song">Song Library</b-dropdown-item>
            <b-dropdown-item href="/song/advancedsearchform" class="nav-subitem"
              >Advanced Search</b-dropdown-item
            >
            <b-dropdown-item href="/song/augment" class="nav-subitem"
              >Add Song</b-dropdown-item
            >
            <b-dropdown-item href="/song/newmusic" class="nav-subitem"
              >New Music</b-dropdown-item
            >
            <b-dropdown-item href="/dances/wedding-music"
              >Wedding</b-dropdown-item
            >
            <b-dropdown-item href="/song/holidaymusic">Holiday</b-dropdown-item>
            <b-dropdown-item href="/song/holidaymusic?occassion=halloween"
              >Halloween</b-dropdown-item
            >
            <b-dropdown-item href="/tag">Tags</b-dropdown-item>
          </b-nav-item-dropdown>
          <b-nav-item-dropdown id="tools-menu" text="Tools">
            <b-dropdown-item href="/home/counter"
              >Tempo Counter</b-dropdown-item
            >
            <b-dropdown-item href="/home/tempi">Tempi (Tempos)</b-dropdown-item>
            <b-dropdown-item href="/song/advancedsearchform"
              >Advanced Search</b-dropdown-item
            >
            <b-dropdown-item
              v-if="context.isBeta || context.isAdmin"
              href="/home/spotifyexplorer"
              >Spotify Explorer (BETA)</b-dropdown-item
            >
          </b-nav-item-dropdown>
          <b-nav-item-dropdown text="Info">
            <b-dropdown-item :href="context.helpLink">Help</b-dropdown-item>
            <b-dropdown-item href="https://music4dance.blog/"
              >Blog</b-dropdown-item
            >
            <b-dropdown-item href="/home/faq">FAQ</b-dropdown-item>
            <b-dropdown-item href="/home/about"
              >About music4dance</b-dropdown-item
            >
            <b-dropdown-item href="/home/readinglist"
              >Reading List</b-dropdown-item
            >
            <b-dropdown-item href="/home/sitemap">Site Map</b-dropdown-item>
            <b-dropdown-item href="/home/privacypolicy"
              >Privacy Policy</b-dropdown-item
            >
            <b-dropdown-item href="/home/credits">Credits</b-dropdown-item>
          </b-nav-item-dropdown>
          <b-nav-item id="contribute-menu" right href="/home/contribute"
            >Contribute</b-nav-item
          >
          <b-nav-item-dropdown
            id="admin-menu"
            text="Admin"
            v-if="context.isAdmin"
          >
            <b-dropdown-item href="/admin">Index</b-dropdown-item>
            <b-dropdown-item href="/applicationusers">Users</b-dropdown-item>
            <b-dropdown-item href="/activitylog">Activity Log</b-dropdown-item>
            <b-dropdown-item href="/tag/list">Tags</b-dropdown-item>
            <b-dropdown-item href="/admin/diagnostics"
              >Diagnostics</b-dropdown-item
            >
            <b-dropdown-item href="/admin/initializationTasks"
              >Initialization and Cleanup</b-dropdown-item
            >
            <b-dropdown-item href="/admin/scraping">Scraping</b-dropdown-item>
            <b-dropdown-item href="/playlist">PlayLists</b-dropdown-item>
            <b-dropdown-item href="/Searches?showDetails=True&user=all"
              >Searches</b-dropdown-item
            >
            <b-dropdown-item href="/song/rawsearchform"
              >Raw Search</b-dropdown-item
            >
            <b-dropdown-item href="/admin/uploadbackup"
              >Uploads and Backups</b-dropdown-item
            >
          </b-nav-item-dropdown>
        </b-navbar-nav>

        <b-navbar-nav class="ml-auto">
          <b-nav-form @submit="search">
            <b-form-input
              v-model="searchString"
              size="sm"
              class="mr-sm-2"
              placeholder="Search"
              @input="checkServiceAndWarn"
            ></b-form-input>
            <b-button
              size="sm"
              class="my-2 my-sm-0"
              type="submit"
              :disabled="!searchString.trim()"
              >Search</b-button
            >
          </b-nav-form>
          <template v-if="context.userName">
            <b-nav-item-dropdown :html="profileHeader">
              <b-dropdown-item right href="/identity/account/manage"
                >My Profile</b-dropdown-item
              >
              <b-dropdown-item right :href="songLink">My Songs</b-dropdown-item>
              <b-dropdown-item right :href="searchLink"
                >My Searches</b-dropdown-item
              >
              <b-dropdown-item
                right
                href="javascript:document.getElementById('logoutForm').submit()"
                >Log out</b-dropdown-item
              >
            </b-nav-item-dropdown>
          </template>
          <template v-else>
            <b-nav-item right :href="accountLink('register')"
              >Register</b-nav-item
            >
            <b-nav-item right :href="accountLink('login')">Login</b-nav-item>
          </template>
        </b-navbar-nav>
      </b-collapse>
    </b-navbar>
    <b-alert
      v-if="showMarketing"
      show
      variant="success"
      style="margin-bottom: 0"
      dismissible
      @dismissed="onDismissed('marketing-acknowledged')"
      ><span v-html="context.marketingMessage"></span
    ></b-alert>
    <b-alert
      variant="warning"
      show
      dismissible
      v-if="showExpiration"
      style="margin-bottom: 0"
      @dismissed="onDismissed('renewal-acknowledged')"
    >
      Your premium subcription will expire in
      {{ Math.round(context.daysToExpiration || 0) }}
      day(s). Please
      <a href="/home/contribute" class="alert-link">click here</a> to renew.
      Thanks for your continued support.
    </b-alert>
    <form
      id="logoutForm"
      action="/identity/account/logout"
      method="post"
      style="height: 0"
    >
      <input
        name="__RequestVerificationToken"
        type="hidden"
        :value="context.xsrfToken"
      />
      <input type="hidden" name="returnUrl" value="/" />
      <button id="logout" type="submit" class="btn btn-link"></button>
    </form>
  </div>
</template>

<script lang="ts">
import DropTarget from "@/mix-ins/DropTarget";
import { MenuContext } from "@/model/MenuContext";
import "reflect-metadata";
import { PropType } from "vue";

const renewal = "renewal-acknowledged";
const marketing = "marketing-acknowledged";

export default DropTarget.extend({
  props: {
    context: { type: Object as PropType<MenuContext>, required: true },
  },
  data() {
    return new (class {
      searchString = "";
    })();
  },
  computed: {
    profileHeader(): string {
      const context = this.context;
      const index =
        context.hasRole("dbAdmin") && context.indexId
          ? ` (${context.indexId})`
          : "";
      return `${context.userName}${index} <img src="/images/swing-ui.png" alt="User Icon" height="30" width="30" />`;
    },

    songLink(): string {
      return `/users/info/${this.context.userName}`;
    },

    searchLink(): string {
      return `/searches/?user=${this.context.userName}`;
    },

    showExpiration(): boolean {
      return (
        this.context.daysToExpiration !== undefined &&
        this.context.daysToExpiration < 30 &&
        !sessionStorage.getItem(renewal)
      );
    },

    showMarketing(): boolean {
      return (
        !!this.context.marketingMessage && !sessionStorage.getItem(marketing)
      );
    },

    isTest(): boolean {
      return window.location.hostname.endsWith(".azurewebsites.net");
    },
  },
  methods: {
    accountLink(type: string): string {
      const url = window.location.pathname + window.location.search;
      return `/identity/account/${type}?returnUrl=${url}`;
    },
    onDismissed(name: string): void {
      sessionStorage.setItem(name, "true");
    },
    search(): void {
      event?.preventDefault();
      window.location.href = `/search?search=${encodeURIComponent(
        this.searchString
      )}`;
    },
  },
});
</script>
