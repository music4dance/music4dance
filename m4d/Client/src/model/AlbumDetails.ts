import "reflect-metadata";
import { jsonMember, jsonObject } from "typedjson";
import { PurchaseEncoded } from "./Purchase";

@jsonObject
export class AlbumDetails {
  @jsonMember public index?: number;
  @jsonMember public name?: string;
  @jsonMember public track?: number;
  @jsonMember public publisher?: string;
  @jsonMember public purchase!: PurchaseEncoded;

  public constructor(init?: Partial<AlbumDetails>) {
    Object.assign(this, init);
    if (!this.purchase) {
      this.purchase = new PurchaseEncoded();
    }
  }
}
