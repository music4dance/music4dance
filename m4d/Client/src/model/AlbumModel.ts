import { jsonMember, jsonObject } from "typedjson";
import { SongListModel } from "./SongListModel";

@jsonObject
export class AlbumModel extends SongListModel {
  @jsonMember public title!: string;
  @jsonMember public artist?: string;
}
