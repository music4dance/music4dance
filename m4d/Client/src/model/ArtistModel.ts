import "reflect-metadata";
import { jsonMember, jsonObject } from "typedjson";
import { SongListModel } from "./SongListModel";

@jsonObject
export class ArtistModel extends SongListModel {
  @jsonMember public artist!: string;
}
