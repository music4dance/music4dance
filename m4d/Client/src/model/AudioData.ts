import "reflect-metadata";
import { jsonMember, jsonObject } from "typedjson";

@jsonObject
export class AudioData {
  @jsonMember public beatsPerMeasure?: number;
  @jsonMember public beatsPerMinute?: number;
  @jsonMember public danceability?: number;
  @jsonMember public energy?: number;
  @jsonMember public valence?: number;
}
