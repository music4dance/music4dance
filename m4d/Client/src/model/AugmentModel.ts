import "reflect-metadata";
import { jsonMember, jsonObject } from "typedjson";

@jsonObject
export class AugmentModel {
  @jsonMember public title?: string;
  @jsonMember public artist?: string;
  @jsonMember public id?: string;
}
