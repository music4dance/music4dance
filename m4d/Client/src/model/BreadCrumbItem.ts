export interface BreadCrumbItem {
  text: string;
  href?: string;
  active?: boolean;
}

export const homeCrumb: BreadCrumbItem = { text: "music4dance", href: "/" };

export const danceCrumb: BreadCrumbItem = { text: "Dances", href: "/dances" };

export const songCrumb: BreadCrumbItem = {
  text: "Song Library",
  href: "/song",
};

export const ballroomCrumb: BreadCrumbItem = {
  text: "Ballroom",
  href: "/dances/ballroom-competition-categories",
};

export const infoCrumb: BreadCrumbItem = { text: "Info", href: "/home/info" };

export const danceTrail: BreadCrumbItem[] = [homeCrumb, danceCrumb];

export const ballroomTrail: BreadCrumbItem[] = [
  homeCrumb,
  danceCrumb,
  ballroomCrumb,
];

export const infoTrail: BreadCrumbItem[] = [homeCrumb, infoCrumb];
