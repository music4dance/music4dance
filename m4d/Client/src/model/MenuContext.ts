export interface MenuContext {
  helpLink?: string;
  userName?: string;
  isAdmin?: boolean;
  indexId?: string;
  xsrfToken?: string;
}
