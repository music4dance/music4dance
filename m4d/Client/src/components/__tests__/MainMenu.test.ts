import { createLocalVue, mount } from "@vue/test-utils";
import BootstrapVue from "bootstrap-vue";
import MainMenu from "../MainMenu.vue";
import { MenuContext } from "../../model/MenuContext";

describe("MainMenu.vue", () => {
  test("renders the main menu with default options", () => {
    const localVue = createLocalVue();
    localVue.use(BootstrapVue);

    const context: MenuContext = {};

    const wrapper = mount(MainMenu, {
      localVue,
      propsData: { context },
    });

    expect(wrapper).toMatchSnapshot();
  });

  test("renders the main menu with admin options", () => {
    const localVue = createLocalVue();
    localVue.use(BootstrapVue);
    const context: MenuContext = { userName: "dwgray", isAdmin: true };

    const wrapper = mount(MainMenu, {
      localVue,
      propsData: { context },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
