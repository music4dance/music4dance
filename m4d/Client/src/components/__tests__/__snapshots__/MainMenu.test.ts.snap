// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu.vue renders the main menu with admin options 1`] = `
<div>
  <!---->
  <!---->
  <nav class="navbar navbar-dark bg-primary navbar-expand-lg" id="mainMenu"><a href="/" target="_self" class="navbar-brand"><img src="/images/header-logo.png" height="40" title="music4dance"></a> <button type="button" aria-label="Toggle navigation" class="navbar-toggler collapsed" aria-expanded="false" aria-controls="nav-collapse"><span class="navbar-toggler-icon"></span></button>
    <transition-stub css="true" enterclass="" leaveclass="collapse show" entertoclass="collapse show" leavetoclass="collapse" enteractiveclass="collapsing" leaveactiveclass="collapsing">
      <div id="nav-collapse" class="navbar-collapse collapse" style="display: none;">
        <ul class="navbar-nav">
          <li id="music-menu" class="nav-item b-nav-dropdown dropdown"><a id="music-menu__BV_toggle_" role="button" aria-haspopup="true" aria-expanded="false" aria-controls="music-menu__BV_toggle_menu_" href="#music-menu" target="_self" class="nav-link dropdown-toggle"><span>Music</span></a>
            <ul tabindex="-1" aria-labelledby="music-menu__BV_toggle_" id="music-menu__BV_toggle_menu_" class="dropdown-menu">
              <li role="presentation"><a role="menuitem" href="/dances" target="_self" class="dropdown-item">Dances</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/dances/ballroom-competition-categories" target="_self" class="dropdown-item">Ballroom</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/dances/latin" target="_self" class="dropdown-item">Latin</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/dances/swing" target="_self" class="dropdown-item">Swing</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/dances/tango" target="_self" class="dropdown-item">Tango</a></li>
              <li role="presentation"><a role="menuitem" href="/song" target="_self" class="dropdown-item">Song Library</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/song/advancedsearchform" target="_self" class="dropdown-item">Advanced Search</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/song/augment" target="_self" class="dropdown-item">Add Song</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/song/newmusic" target="_self" class="dropdown-item">New Music</a></li>
              <li role="presentation"><a role="menuitem" href="/dances/wedding-music" target="_self" class="dropdown-item">Wedding</a></li>
              <li role="presentation"><a role="menuitem" href="/song/holidaymusic" target="_self" class="dropdown-item">Holiday</a></li>
              <li role="presentation"><a role="menuitem" href="/song/holidaymusic?occassion=halloween" target="_self" class="dropdown-item">Halloween</a></li>
              <li role="presentation"><a role="menuitem" href="/tag" target="_self" class="dropdown-item">Tags</a></li>
            </ul>
          </li>
          <li id="tools-menu" class="nav-item b-nav-dropdown dropdown"><a id="tools-menu__BV_toggle_" role="button" aria-haspopup="true" aria-expanded="false" aria-controls="tools-menu__BV_toggle_menu_" href="#tools-menu" target="_self" class="nav-link dropdown-toggle"><span>Tools</span></a>
            <ul tabindex="-1" aria-labelledby="tools-menu__BV_toggle_" id="tools-menu__BV_toggle_menu_" class="dropdown-menu">
              <li role="presentation"><a role="menuitem" href="/home/counter" target="_self" class="dropdown-item">Tempo Counter</a></li>
              <li role="presentation"><a role="menuitem" href="/home/tempi" target="_self" class="dropdown-item">Tempi (Tempos)</a></li>
              <li role="presentation"><a role="menuitem" href="/song/advancedsearchform" target="_self" class="dropdown-item">Advanced Search</a></li>
              <li role="presentation"><a role="menuitem" href="/home/spotifyexplorer" target="_self" class="dropdown-item">Spotify Explorer (BETA)</a></li>
            </ul>
          </li>
          <li class="nav-item b-nav-dropdown dropdown"><a role="button" aria-haspopup="true" aria-expanded="false" href="#" target="_self" class="nav-link dropdown-toggle"><span>Info</span></a>
            <ul tabindex="-1" class="dropdown-menu">
              <li role="presentation"><a role="menuitem" href="#" target="_self" class="dropdown-item">Help</a></li>
              <li role="presentation"><a role="menuitem" href="https://music4dance.blog/" target="_self" class="dropdown-item">Blog</a></li>
              <li role="presentation"><a role="menuitem" href="/home/faq" target="_self" class="dropdown-item">FAQ</a></li>
              <li role="presentation"><a role="menuitem" href="/home/about" target="_self" class="dropdown-item">About music4dance</a></li>
              <li role="presentation"><a role="menuitem" href="/home/readinglist" target="_self" class="dropdown-item">Reading List</a></li>
              <li role="presentation"><a role="menuitem" href="/home/sitemap" target="_self" class="dropdown-item">Site Map</a></li>
              <li role="presentation"><a role="menuitem" href="/home/privacypolicy" target="_self" class="dropdown-item">Privacy Policy</a></li>
              <li role="presentation"><a role="menuitem" href="/home/credits" target="_self" class="dropdown-item">Credits</a></li>
            </ul>
          </li>
          <li id="contribute-menu" right="" class="nav-item"><a href="/home/contribute" target="_self" class="nav-link">Contribute</a></li>
          <li id="admin-menu" class="nav-item b-nav-dropdown dropdown"><a id="admin-menu__BV_toggle_" role="button" aria-haspopup="true" aria-expanded="false" aria-controls="admin-menu__BV_toggle_menu_" href="#admin-menu" target="_self" class="nav-link dropdown-toggle"><span>Admin</span></a>
            <ul tabindex="-1" aria-labelledby="admin-menu__BV_toggle_" id="admin-menu__BV_toggle_menu_" class="dropdown-menu">
              <li role="presentation"><a role="menuitem" href="/admin" target="_self" class="dropdown-item">Index</a></li>
              <li role="presentation"><a role="menuitem" href="/applicationusers" target="_self" class="dropdown-item">Users</a></li>
              <li role="presentation"><a role="menuitem" href="/activitylog" target="_self" class="dropdown-item">Activity Log</a></li>
              <li role="presentation"><a role="menuitem" href="/tag/list" target="_self" class="dropdown-item">Tags</a></li>
              <li role="presentation"><a role="menuitem" href="/admin/diagnostics" target="_self" class="dropdown-item">Diagnostics</a></li>
              <li role="presentation"><a role="menuitem" href="/admin/initializationTasks" target="_self" class="dropdown-item">Initialization and Cleanup</a></li>
              <li role="presentation"><a role="menuitem" href="/admin/scraping" target="_self" class="dropdown-item">Scraping</a></li>
              <li role="presentation"><a role="menuitem" href="/playlist" target="_self" class="dropdown-item">PlayLists</a></li>
              <li role="presentation"><a role="menuitem" href="/Searches?showDetails=True&amp;user=all" target="_self" class="dropdown-item">Searches</a></li>
              <li role="presentation"><a role="menuitem" href="/song/rawsearchform" target="_self" class="dropdown-item">Raw Search</a></li>
              <li role="presentation"><a role="menuitem" href="/admin/uploadbackup" target="_self" class="dropdown-item">Uploads and Backups</a></li>
            </ul>
          </li>
        </ul>
        <ul class="navbar-nav ml-auto">
          <li class="form-inline">
            <form class="form-inline"><input type="text" placeholder="Search" class="mr-sm-2 form-control form-control-sm"> <button type="submit" disabled="disabled" class="btn my-2 my-sm-0 btn-secondary btn-sm disabled">Search</button></form>
          </li>
          <li class="nav-item b-nav-dropdown dropdown"><a role="button" aria-haspopup="true" aria-expanded="false" href="#" target="_self" class="nav-link dropdown-toggle"><span>dwgray <img src="/images/swing-ui.png" alt="User Icon" height="30" width="30"></span></a>
            <ul tabindex="-1" class="dropdown-menu">
              <li role="presentation"><a right="" role="menuitem" href="/identity/account/manage" target="_self" class="dropdown-item">My Profile</a></li>
              <li role="presentation"><a right="" role="menuitem" href="/users/info/dwgray" target="_self" class="dropdown-item">My Songs</a></li>
              <li role="presentation"><a right="" role="menuitem" href="/searches/?user=dwgray" target="_self" class="dropdown-item">My Searches</a></li>
              <li role="presentation"><a right="" role="menuitem" href="javascript:document.getElementById('logoutForm').submit()" target="_self" class="dropdown-item">Log out</a></li>
            </ul>
          </li>
        </ul>
      </div>
    </transition-stub>
  </nav>
  <!---->
  <!---->
  <form id="logoutForm" action="/identity/account/logout" method="post" style="height: 0px;"><input name="__RequestVerificationToken" type="hidden"> <input type="hidden" name="returnUrl" value="/"> <button id="logout" type="submit" class="btn btn-link"></button></form>
</div>
`;

exports[`MainMenu.vue renders the main menu with default options 1`] = `
<div>
  <!---->
  <!---->
  <nav class="navbar navbar-dark bg-primary navbar-expand-lg" id="mainMenu"><a href="/" target="_self" class="navbar-brand"><img src="/images/header-logo.png" height="40" title="music4dance"></a> <button type="button" aria-label="Toggle navigation" class="navbar-toggler collapsed" aria-expanded="false" aria-controls="nav-collapse"><span class="navbar-toggler-icon"></span></button>
    <transition-stub css="true" enterclass="" leaveclass="collapse show" entertoclass="collapse show" leavetoclass="collapse" enteractiveclass="collapsing" leaveactiveclass="collapsing">
      <div id="nav-collapse" class="navbar-collapse collapse" style="display: none;">
        <ul class="navbar-nav">
          <li id="music-menu" class="nav-item b-nav-dropdown dropdown"><a id="music-menu__BV_toggle_" role="button" aria-haspopup="true" aria-expanded="false" aria-controls="music-menu__BV_toggle_menu_" href="#music-menu" target="_self" class="nav-link dropdown-toggle"><span>Music</span></a>
            <ul tabindex="-1" aria-labelledby="music-menu__BV_toggle_" id="music-menu__BV_toggle_menu_" class="dropdown-menu">
              <li role="presentation"><a role="menuitem" href="/dances" target="_self" class="dropdown-item">Dances</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/dances/ballroom-competition-categories" target="_self" class="dropdown-item">Ballroom</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/dances/latin" target="_self" class="dropdown-item">Latin</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/dances/swing" target="_self" class="dropdown-item">Swing</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/dances/tango" target="_self" class="dropdown-item">Tango</a></li>
              <li role="presentation"><a role="menuitem" href="/song" target="_self" class="dropdown-item">Song Library</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/song/advancedsearchform" target="_self" class="dropdown-item">Advanced Search</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/song/augment" target="_self" class="dropdown-item">Add Song</a></li>
              <li role="presentation" class="nav-subitem"><a role="menuitem" href="/song/newmusic" target="_self" class="dropdown-item">New Music</a></li>
              <li role="presentation"><a role="menuitem" href="/dances/wedding-music" target="_self" class="dropdown-item">Wedding</a></li>
              <li role="presentation"><a role="menuitem" href="/song/holidaymusic" target="_self" class="dropdown-item">Holiday</a></li>
              <li role="presentation"><a role="menuitem" href="/song/holidaymusic?occassion=halloween" target="_self" class="dropdown-item">Halloween</a></li>
              <li role="presentation"><a role="menuitem" href="/tag" target="_self" class="dropdown-item">Tags</a></li>
            </ul>
          </li>
          <li id="tools-menu" class="nav-item b-nav-dropdown dropdown"><a id="tools-menu__BV_toggle_" role="button" aria-haspopup="true" aria-expanded="false" aria-controls="tools-menu__BV_toggle_menu_" href="#tools-menu" target="_self" class="nav-link dropdown-toggle"><span>Tools</span></a>
            <ul tabindex="-1" aria-labelledby="tools-menu__BV_toggle_" id="tools-menu__BV_toggle_menu_" class="dropdown-menu">
              <li role="presentation"><a role="menuitem" href="/home/counter" target="_self" class="dropdown-item">Tempo Counter</a></li>
              <li role="presentation"><a role="menuitem" href="/home/tempi" target="_self" class="dropdown-item">Tempi (Tempos)</a></li>
              <li role="presentation"><a role="menuitem" href="/song/advancedsearchform" target="_self" class="dropdown-item">Advanced Search</a></li>
              <!---->
            </ul>
          </li>
          <li class="nav-item b-nav-dropdown dropdown"><a role="button" aria-haspopup="true" aria-expanded="false" href="#" target="_self" class="nav-link dropdown-toggle"><span>Info</span></a>
            <ul tabindex="-1" class="dropdown-menu">
              <li role="presentation"><a role="menuitem" href="#" target="_self" class="dropdown-item">Help</a></li>
              <li role="presentation"><a role="menuitem" href="https://music4dance.blog/" target="_self" class="dropdown-item">Blog</a></li>
              <li role="presentation"><a role="menuitem" href="/home/faq" target="_self" class="dropdown-item">FAQ</a></li>
              <li role="presentation"><a role="menuitem" href="/home/about" target="_self" class="dropdown-item">About music4dance</a></li>
              <li role="presentation"><a role="menuitem" href="/home/readinglist" target="_self" class="dropdown-item">Reading List</a></li>
              <li role="presentation"><a role="menuitem" href="/home/sitemap" target="_self" class="dropdown-item">Site Map</a></li>
              <li role="presentation"><a role="menuitem" href="/home/privacypolicy" target="_self" class="dropdown-item">Privacy Policy</a></li>
              <li role="presentation"><a role="menuitem" href="/home/credits" target="_self" class="dropdown-item">Credits</a></li>
            </ul>
          </li>
          <li id="contribute-menu" right="" class="nav-item"><a href="/home/contribute" target="_self" class="nav-link">Contribute</a></li>
          <!---->
        </ul>
        <ul class="navbar-nav ml-auto">
          <li class="form-inline">
            <form class="form-inline"><input type="text" placeholder="Search" class="mr-sm-2 form-control form-control-sm"> <button type="submit" disabled="disabled" class="btn my-2 my-sm-0 btn-secondary btn-sm disabled">Search</button></form>
          </li>
          <li right="" class="nav-item"><a href="/identity/account/register?returnUrl=/" target="_self" class="nav-link">Register</a></li>
          <li right="" class="nav-item"><a href="/identity/account/login?returnUrl=/" target="_self" class="nav-link">Login</a></li>
        </ul>
      </div>
    </transition-stub>
  </nav>
  <!---->
  <!---->
  <form id="logoutForm" action="/identity/account/logout" method="post" style="height: 0px;"><input name="__RequestVerificationToken" type="hidden"> <input type="hidden" name="returnUrl" value="/"> <button id="logout" type="submit" class="btn btn-link"></button></form>
</div>
`;
